x-pack/test/security_solution_api_integration/test_suites/detections_response/utils/retry_delete_by_query_conflicts.ts

```diff
@@ -18,34 +18,29 @@ const RETRY_DELAY = 200;
  * Retry an Elasticsearch deleteByQuery operation if it runs into 409 Conflicts,
  * up to a maximum number of attempts.
  */
-export async function retryIfDeleteByQueryConflicts<T>(
+export async function retryIfDeleteByQueryConflicts(
   logger: ToolingLog,
   name: string,
   operation: () => Promise<DeleteByQueryResponse>,
   retries: number = RETRY_ATTEMPTS,
   retryDelay: number = RETRY_DELAY
 ): Promise<DeleteByQueryResponse> {
-  const operationResult = await operation();
-  if (!operationResult.failures || operationResult.failures?.length === 0) {
-    return operationResult;
-  }
+  for (let retriesLeft = retries; retriesLeft > 0; retriesLeft--) {
+    const operationResult = await operation();
 
-  for (const failure of operationResult.failures) {
-    if (failure.status === 409) {
-      // if no retries left, throw it
-      if (retries <= 0) {
-        logger.error(`${name} conflict, exceeded retries`);
-        throw new Error(`${name} conflict, exceeded retries`);
-      }
-
-      // Otherwise, delay a bit before retrying
-      logger.debug(`${name} conflict, retrying ...`);
-      await waitBeforeNextRetry(retryDelay);
-      return await retryIfDeleteByQueryConflicts(logger, name, operation, retries - 1);
+    if (!operationResult.failures || operationResult.failures?.length === 0) {
+      logger.info(`${name} finished successfully`);
+      return operationResult;
     }
+
+    const failureCause = operationResult.failures.map((failure) => failure.cause).join(', ');
+
+    logger.warning(`Unable to delete by query ${name} caused by ${failureCause}, retrying ...`);
```

Comment:
nit: I would reformat this to something more structured like:

`logger.warning('Unable to delete by query ${name}. Caused by: "${failureCause}". Retrying ...');`

Since `failureCause` might be a whole sentence (or concatenation of sentences) it will probably be hard to mentally parse it while reading the printed log if it is added in the middle of the log message,